atomic_reactor/plugins/post_group_manifests.py

```diff
@@ -318,62 +154,32 @@ def sort_annotations(self):
-                repos.append(digest)
-
-        sources = {}
-        for registry in self.registries:
-            hostname = registry_hostname(registry)
-            platforms = sorted_digests.get(hostname, {})
-
-            if set(platforms) != all_platforms:
-                raise RuntimeError("Missing platforms for registry {}: found {}, expected {}"
-                                   .format(registry, sorted(platforms), sorted(all_platforms)))
-
-            selected_digests = {}
-            for p, repos in platforms.items():
-                selected_digests[p] = sorted(repos, key=lambda d: d['repository'])[0]
+        return self.manifest_util.sort_annotations(all_annotations)
 
-            sources[registry] = selected_digests
-
-        return sources
-
-    def get_registry_session(self, registry):
-        registry_conf = self.registries[registry]
-
-        insecure = registry_conf.get('insecure', False)
-        secret_path = registry_conf.get('secret')
-
-        return RegistrySession(registry, insecure=insecure,
-                               dockercfg_path=secret_path,
-                               access=('pull', 'push'))
+    def tag_manifest_into_registry(self, session, source_digest, source_repo, images):
+        manifest, media, digest = self.manifest_util.tag_manifest_into_registry(session,
+                                                                                source_digest,
+                                                                                source_repo,
+                                                                                images)
+        return {'manifest': manifest, 'media_type': media, 'manifest_digest': digest}
 
     def run(self):
-        digests = dict()
+        primary_images = get_primary_images(self.workflow)
+        unique_images = get_unique_images(self.workflow)
+        self.non_floating_images = primary_images + unique_images
+
         for registry, source in self.sort_annotations().items():
-            session = self.get_registry_session(registry)
+            session = self.manifest_util.get_registry_session(registry)
 
             if self.group:
-                repo, digest = self.group_manifests_and_tag(session, source)
-                self.log.debug("repo: %s digest: %s", repo, digest)
-                digests[repo] = digest
+                return self.group_manifests_and_tag(session, source)
             else:
-                found = False
                 if len(source) != 1:
                     raise RuntimeError('Without grouping only one source is expected')
-                for digest in source.values():
-                    self.tag_manifest_into_registry(session, digest)
-                    found = True
-                if not found:
-                    raise ValueError('Failed to find any platform')
-        return digests
+                for orig_digest in source.values():
+                    break
```

Comment:
pylint is complaining about using `orig_digest` after this loop. Maybe `orig_digest = source.values()[0]` is a more explicit way of saying that we need there to be values, and we'll just pick the first one we find.

Reply:
source.values() is a `dict_values` object, not a list, and can't be indexed that way.

Made what's happening more explicit so the linter will stop complaining.